base_changeset/models/record_changeset_change.py

```diff
@@ -383,9 +383,12 @@ def get_fields_changeset_changes(self, model, res_id):
 
     def _compute_user_can_validate_changeset(self):
         is_superuser = self.env.user.id == SUPERUSER_ID
-        has_group = self.user_has_groups("base_changeset.group_changeset_user")
+        user_groups = self.env.user.groups_id
         for rec in self:
-            can_validate = rec._is_change_pending() and (is_superuser or has_group)
+            can_validate = rec._is_change_pending() and (
+                is_superuser or
+                rec.rule_id.validator_group_ids & user_groups
```

Comment:
In case field Validators (validator_group_ids) is empty, shouldn't `group_changeset_user` be enabled anyways? Or for example in case `group_changeset_user` is accidentally removed from validator_group_ids.